doc/src/tutorials/physics/mechanics/atwoods_machine_example.rst

```diff
@@ -0,0 +1,385 @@
+by the same amount, so that the total rope length remains constant.
+
+:obj:`sympy.physics.mechanics` provides wrapping geometry classes and a wrapping pathway
+that are useful to model forces along a surface.
+
+We can model the pulley as a ``WrappingCylinder`` and the rope
+as a ``WrappingPathway`` that wraps over this cylinder. This example
+illustrates how to set up the kinematics, compute the total rope length, verify
+inextensibility of the rope, and derive the equations of motion using Kane's method.
+
+Define Variables
+================
+
+First, import the necessary symbols, frames, points, and classes. We introduce
+a single generalized coordinate $q(t)$, which measures the downward
+displacement of mass $m_1$. Its time derivative $u(t) = \dot q(t)$ is the
+generalized speed.
+
+.. plot::
+    :format: doctest
+    :include-source: True
+    :context: reset
+    :nofigs:
+
+    >>> import sympy as sp
+    >>> from sympy import Q, refine
+    >>> from sympy.physics.mechanics import (
+    ...      ReferenceFrame,
+    ...      Point,
+    ...      Particle,
+    ...      KanesMethod,
+    ...      dynamicsymbols,
+    ...      WrappingCylinder,
+    ...      WrappingPathway,
+    ...      Force,
+    ...  )
+    >>>
+    >>> t = sp.symbols("t")
+    >>>
+    >>> # Constant parameters: masses, gravity, pulley radius, initial height, tension
+    >>>
+    >>> m1, m2, g, r, h, T = sp.symbols("m1 m2 g r h T", positive=True, real=True)
+    >>>
+    >>> # Generalized coordinate and speed: m1 moves downward by q(t)
+    >>>
+    >>> q = dynamicsymbols("q", real=True)   # q(t)
+    >>> u = dynamicsymbols("u", real=True)   # u(t) = dq/dt
+
+Inertial Frame and Pulley Center
+================================
+
+We define an inertial frame $N$ and fix the pulley's center $O$ at the
+origin. The pulley axis is aligned with the $N.x$ direction.
+
+.. plot::
+    :format: doctest
+    :include-source: True
+    :context: close-figs
+    :nofigs:
+
+    >>> # Define inertial reference frame and pulley center
+    >>>
+    >>> N = ReferenceFrame("N")
+    >>> O = Point("O")
+    >>> O.set_vel(N, 0)  # Pulley center is fixed at the origin
+
+Positions and Velocities of the Two Masses
+==========================================
+
+Let $P_1$ be the contact point (mass point) of $m_1$ and $P_2$ that of
+$m_2$. Initially, when $q=0$, both masses are at vertical distance $h+r$
+below the center of the pulley. As $m_1$ descends by $q$, its position is
+
+$$
+P_1: (x=0,\; y=+r,\; z=-(h+q))\,,
+$$
+
+and $m_2$ rises accordingly to
+
+$$
+P_2: (x=0,\; y=-r,\; z=-(h-q))\,.
+$$
+
+We set their velocities by differentiating their position vectors in frame
+$N$.
+
+.. plot::
+    :format: doctest
+    :include-source: True
+    :context: close-figs
+    :nofigs:
+
+    >>> # Mass m1 at P1: (x=0, y=+r, z=-(h+q))
+    >>>
+    >>> P1 = Point("P1")
+    >>> P1.set_pos(O, r * N.y + (-(h + q)) * N.z)
+    >>> P1.set_vel(N, P1.pos_from(O).diff(t, N))
```

Comment:
P1.vel(N) will do this differentiation internally. There is no need to set velocities unless you need them to be something different.